modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -725,10 +758,27 @@ namespace cv {
                             tensor_shape[0] += net->out_channels_vec[layers_vec[k]];
                         }
 
-                        if (layers_vec.size() == 1)
-                            setParams.setIdentity(layers_vec.at(0));
+                        if (num_splits > 1)
+                        {
+                            int group_id = getParam<int>(layer_params, "group_id");
+                            tensor_shape[0] /= num_splits;
+                            int split_size = tensor_shape[0] / layers_vec.size();
+                            setParams.setSlice(layers_vec.size(), layers_vec.data(), split_size, group_id);
+                            if (layers_vec.size() > 1)
+                            {
+                                for (size_t k = 0; k < layers_vec.size(); ++k)
+                                    layers_vec[k] += layers_vec.size();
```

Comment:
For which layers we should increment indices?

Reply:
For Concat after Slice

Reply:
But why we have a single Slice layer but have to increase by number of it's outputs?

Reply:
We increment layer ids, not number of outputs

Reply:
Yes, but it;s not clear for me that for a single layer (Slice) we need to add N indices (by number of Concat inputs?). I believe some comment in code about layer behavior will help to explain this.